exonum/src/proto/mod.rs

```diff
@@ -335,6 +337,77 @@ where
     }
 }
 
+impl<K, V> ProtobufConvert for exonum_merkledb::MapProof<K, V>
+where
+    K: BinaryKey,
```

Comment:
I'd suggest to restrict `key`s to types implementing `ProtobufConvert` (meaning "any type we know how to (de)serialize to / from Protobuf") instead of `BinaryKey`, and use the corresponding (de)serialization logic. The less proofs expose internal logic of MerkleDB, the better.